doc/src/challenges/0023-vec-pt1.md

```diff
@@ -0,0 +1,81 @@
+# Challenge 17: Verify the safety of `Vec` functions part 1
+
+- **Status:** Open
+- **Tracking Issue:** [#29](https://github.com/model-checking/verify-rust-std/issues/29)
+- **Start date:** *2025-03-07*
+- **End date:** *2025-10-17*
+- **Reward:** *?*
+
+-------------------
+
+
+## Goal
+
+Verify the memory safety of [`std::Vec`] functions (library/alloc/src/vec/mod.rs).
```

Comment:
Is it really just "memory" safety, or can we perhaps do other properties as well?